dbms/src/Storages/DeltaMerge/File/ColumnStream.cpp

```diff
@@ -0,0 +1,316 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/DeltaMerge/File/ColumnStream.h>
+#include <Storages/DeltaMerge/File/DMFileReader.h>
+#include <Storages/Page/PageUtil.h>
+
+namespace DB::DM
+{
+// TODO: make `MarkLoader` as a part of DMFile?
+class MarkLoader
+{
+public:
+    MarksInCompressedFilePtr operator()()
```

Comment:
Maybe define as a function rather than a operator to improve readability?

Reply:
It need to be callable object for `reader.mark_cache->getOrSet(..., MarkLoader())`

Reply:
ok, add some comments to explain where it is be used?